src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> strategiesMap;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> strategiesMap) {
+        this.strategiesMap = strategiesMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(Operation operation) {
+        String exceptionMessage =
+                "Cannot find OperationHandler for operation = [" + operation + "]";
+        return Optional.ofNullable(strategiesMap.get(operation))
+                .orElseThrow(() -> new NoSuchElementException(exceptionMessage));
```

Comment:
Avoid using `Optional` for simply checking if a value is present or not. A more conventional approach would be to use a direct null check and throw the exception if needed. Using `Optional` here is an unnecessary overhead. Consider refactoring to a simple null check.